### DISCUSSION TITLE
How to set up PHPStorm when stubs have multiple class definitions inside if conditions?

### DISCUSSION THREAD

**Discussion by michnovka** [doctrine/DoctrineBundle#1619] [Q&A]
I am using PHPStorm for my projects, and recently I come to an issue with

```php
/**
 * @extends ServiceEntityRepository<TournamentPlayerStats>
 */
class TournamentPlayerStatsRepository extends ServiceEntityRepository
{
```

PHPStorm complains that `ServiceEntityRepository` has multiple definitions.

This is true, as the stubs in this bundle contain an if check and define the class on 2 lines:

https://github.com/doctrine/DoctrineBundle/blob/2.8.x/Repository/ServiceEntityRepository.php#L18
and
https://github.com/doctrine/DoctrineBundle/blob/2.8.x/Repository/ServiceEntityRepository.php#L39

based on whether trait `LazyGhostTrait` exists. So how can I "explain" to PHPStorm to use just one definition?

**Comment by stof** [comment]
That's a question that should be asked to Jetbrains, not to us.

**Comment by ostrolucky** [comment]
This problem was solved in 3.x
